### PR TITLE
fix for NUTCH-2460 contributed by Hussein Alahmad

### PR DESCRIPTION
**Pull request by hussein-alahmad**
use the headless option of firefox and chrome in protocol-selenium

the --headless option is added to firefox in version 55 or later , and in chrome in version 59 or later ...
this is much better than relying on xvfb and its associates .
we can add it as a property in the config file .
I'm trying it on my local machine , and will create a pull request when I finish testing it 

I've tested it using firefox 57.0 , gecodriver 0.19.1 and selenium 3.7.1

Important note : you need to add the following property to nutch-default.xml or nutch-site.xml for the headless option to work

<property>
        <name>selenium.firefox.headless</name>
        <value>true</value>
        <description>A Boolean value representing if firefox should
            run headless . make sure that firefox version is 55 or later,
            and selenium webDriver version is 3.6.0 or later. The default value is false.
            Currently this option exist for - 'firefox' </description>
</property>

### CODE REVIEW

File: src/plugin/lib-selenium/src/java/org/apache/nutch/protocol/selenium/HttpWebClient.java [hunk 1]
```diff
@@ -166,7 +175,7 @@ public static String getHTMLContent(WebDriver driver, Configuration conf) {
   public static void cleanUpDriver(WebDriver driver) {
     if (driver != null) {
       try {
-	      driver.close();
+//	      driver.close();
```
Comment (lewismc) [inline src/plugin/lib-selenium/src/java/org/apache/nutch/protocol/selenium/HttpWebClient.java]: @hussein-alahmad Why is this commented?

Comment (hussein-alahmad) [reply src/plugin/lib-selenium/src/java/org/apache/nutch/protocol/selenium/HttpWebClient.java]: there is a driver.quit() statement after it , I was getting an error about trying to close a non existing connection , after searching about it , turned out that driver.quit() alone without the driver.close() do the work 
as mentioned [here](https://github.com/mozilla/geckodriver/issues/966)

**Comment by hussein-alahmad** [comment]
<property>
    <name>selenium.firefox.headless</name>
    <value>false</value>
    <description>A Boolean value representing if firefox should
      run headless . make sure that firefox version is 55 or later,
      and selenium webDriver version is 3.6.0 or later. The default value is false.
      Currently this option exist for - 'firefox' </description>
  </property>

@lewismc
since all the conf files are ignored , should I force adding them ?

File: src/plugin/lib-selenium/src/java/org/apache/nutch/protocol/selenium/HttpWebClient.java (hunk 1 above)
Comment (lewismc) [reply src/plugin/lib-selenium/src/java/org/apache/nutch/protocol/selenium/HttpWebClient.java]: Thank you @hussein-alahmad

**Comment by lewismc** [comment]
@hussein-alahmad ping

**Comment by hussein-alahmad** [comment]
@lewismc  very very sorry for the late reply 
I added the property to nutch-default.xml , and updated the documentation

**Comment by sbatururimi** [comment]
Any plan to merge that?

**Comment by sbatururimi** [comment]
I'm adding more features in NUTCH-2676 also.

**Comment by sebastian-nagel** [comment]
Thanks, @sbatururimi! Let us know whether it still makes sense to merge this PR. Afaics, the requested changes are made and the merge conflicts are easy to resolve.

**Comment by sbatururimi** [comment]
Maybe it will make sense to wait till the mid/end of the next week in order to compare how much have been added :).

**Comment by sebastian-nagel** [comment]
Included in #430 / dfd8602. Thanks, @hussein-alahmad and @sbatururimi!

**Pull request closed without merging**